lib/web_ui/test/engine/semantics/semantics_helper_test.dart

```diff
@@ -0,0 +1,190 @@
+      event = dispatchPointerEvent('pointermove');
+      shouldForwardToFramework =
+          desktopEnableSemantics.tryEnableSemantics(event);
+
+      expect(shouldForwardToFramework, true);
+    });
+
+    test(
+        'Relevants events targeting placeholder should not be forwarded to the framework',
+        () async {
+      // Prework. Attach the placeholder to dom.
+      _placeholder = desktopEnableSemantics.prepareAccesibilityPlaceholder();
+      html.document.body.append(_placeholder);
+
+      html.Event event = dispatchMouseEvent('mousedown', target: _placeholder);
+      bool shouldForwardToFramework =
+          desktopEnableSemantics.tryEnableSemantics(event);
+
+      expect(shouldForwardToFramework, false);
+    });
+
+    test(
+        'After max number of relevant events, events should be forwarded to the framework',
+        () async {
+      // Prework. Attach the placeholder to dom.
+      _placeholder = desktopEnableSemantics.prepareAccesibilityPlaceholder();
+      html.document.body.append(_placeholder);
+
+      html.Event event = dispatchMouseEvent('mousedown', target: _placeholder);
+      bool shouldForwardToFramework =
+          desktopEnableSemantics.tryEnableSemantics(event);
+
+      expect(shouldForwardToFramework, false);
+
+      // Send max number of events;
+      for (int i = 1; i <= kMaxSemanticsActivationAttempts; i++) {
+        event = dispatchPointerEvent('mousedown', target: _placeholder);
+        shouldForwardToFramework =
+            desktopEnableSemantics.tryEnableSemantics(event);
+      }
+
+      expect(shouldForwardToFramework, true);
+    });
+  });
+
+  group('$MobileEnableSemantics', () {
+    MobileEnableSemantics mobileEnableSemantics;
+    html.Element _placeholder;
+
+    setUp(() {
+      mobileEnableSemantics = MobileEnableSemantics();
+    });
+
+    tearDown(() {
+      if (_placeholder != null) {
+        _placeholder.remove();
+      }
+    });
+
+    test('prepare accesibility placeholder', () async {
+      _placeholder = mobileEnableSemantics.prepareAccesibilityPlaceholder();
+
+      expect(_placeholder.getAttribute('role'), 'button');
+
+      html.document.body.append(_placeholder);
+
+      // Placeholder should cover all the screen on a mobile device.
+      final num bodyHeight = html.window.innerHeight;
+      final num bodyWidht = html.window.innerWidth;
+
+      expect(_placeholder.getBoundingClientRect().height, bodyHeight);
+      expect(_placeholder.getBoundingClientRect().width, bodyWidht);
+    });
+
+    test('Not relevant events should be forwarded to the framework', () async {
+      final html.Event event = dispatchTouchEvent('touchcancel');
+      bool shouldForwardToFramework =
+          mobileEnableSemantics.tryEnableSemantics(event);
+
+      expect(shouldForwardToFramework, true);
+    });
+  });
+}
+
+html.PointerEvent dispatchPointerEvent(String type, {html.EventTarget target}) {
+  target ??= html.window;
+
+  final Function jsKeyboardEvent =
```

Comment:
nit: rename `jsKeyboardEvent` :)

Reply:
Actually, I don't think you need to use js interop for this. You can simply use the Dart constructor (i.e. `html.MouseEvent()`). The reason I had to do this for `KeyboardEvent` is because Dart's `html.KeyboardEvent` doesn't expose all the arguments that I needed.

Reply:
thanks!